libqtile/widget/clock.py

```diff
@@ -88,3 +101,30 @@ def poll(self):
         else:
             now = datetime.now(timezone.utc).astimezone()
         return (now + self.DELTA).strftime(self.format)
+
+    @expose_command
+    def update_timezone(self, timezone: str | tzinfo | None = None):
+        """
+        Force the clock to update timezone information.
+
+        If the method is called with no arguments then the widget will reload
+        the timzeone set on the computer (e.g. via ``timedatectl set-timezone ..``).
+        This will have no effect if you have previously set a ``timezone`` value.
+
+        Alternatively, you can pass a timezone string (e.g. ``"Europe/Lisbon"``) to change
+        the specified timezone. Setting this to an empty string will cause the clock
+        to rely on the system timezone.
+        """
+        if timezone:
+            self._check_timezone(timezone)
+        elif timezone is not None:
+            self.timezone = None
+
+        # Force python to update timezone info
+        time.tzset()
```

Comment:
I'm confused about why we need this line. IIUC, either we 1. have an explicit non-none `self.timezone` in which case we pass that in `poll()` when doing our rendering, or 2. the timezone is none, in which case we want to use the system time. Since our rendering code never fiddles with `TZ`, we shouldn't ever need to reset it should we?

Or is the idea that we need this call after a system time change in order to kick python into rendering relative to the new system timezone?

Reply:
Correct. It's the latter.